File: walk-time.go
package main
import ("fmt"
        "math/rand"
        "time"
       )

func main() { 
  rand.Seed(time.Now().UnixNano())

  fmt.Print(rand.Intn(3)+1, ":")
  var mins = rand.Intn(60)
  if mins < 10 { fmt.Print(0) }
  fmt.Print(mins, "PM", "\n")
}

File: walk-time.hs
rand :: Int -> Int
rand seed = (1103515245 * seed + 12345) `mod` (2^32)

main = do
  let num = rand 100
  let hrs = num `mod` 3 + 1
  let mins = num `mod` 60
  
  if mins >= 10
  then print(show hrs ++ ":" ++ show mins ++ "PM")
  else print(show hrs ++ ":0" ++ show mins ++ "PM")
